Adding doc on - How to Use W3C Trace Headers in Dapr

shalabhms opened the pull request:
Adding How to use W3C trace context in Dapr calls.

## Issue reference

Closes #564

orizohar commented on howto/use-w3c-tracecontext/README.md:
```diff
@@ -0,0 +1,243 @@
+# W3C Trace Context - Background
```
Seems like "Background" should be the header for the first section not the entire article

shalabhms replied:
done.

orizohar commented on howto/use-w3c-tracecontext/README.md:
```diff
@@ -0,0 +1,243 @@
+| Node.js | [Link](https://github.com/census-instrumentation/opencensus-node)
+| Python | [Link](https://census-instrumentation.github.io/opencensus-python/trace/api/index.html)
+
+For the purpose of example, we will use the same [grpc app](../create-grpc-app) based on Go SDK.
```
Please rephrase to remove all first person references ("we", "I")

shalabhms replied:
done.

orizohar commented on howto/use-w3c-tracecontext/README.md:
```diff
@@ -0,0 +1,243 @@
+
+So Span is the unit step in a trace. Each span has a name, latency, status and additional metadata.
+
+Below we are starting a span for a cache read and ending it when we are done:
```
Please rephrase to remove all first person references ("we", "I")

orizohar commented on howto/use-w3c-tracecontext/README.md:
```diff
@@ -0,0 +1,243 @@
+
+When you call the Dapr API through HTTP/gRPC, you need to pass the trace context across process. For across the network, OpenCensus provides different propagation methods for different protocols. You can read about the propagation package [here](https://godoc.org/go.opencensus.io/trace/propagation).
+
+In our example, we will use gRPC calls , for that OpenCensus’ binary propagation format is used.
```
Please rephrase to remove all first person references ("we", "I")

orizohar commented on howto/use-w3c-tracecontext/README.md:
```diff
@@ -0,0 +1,243 @@
+Please go here for [Dapr API reference](https://github.com/dapr/docs/tree/master/reference/api).
+
+## Configuring Tracing in Dapr
+As mentioned earlier, in this tracing example, we are using the same [grpc app](../create-grpc-app).
```
Please rephrase to remove all first person references ("we", "I")

orizohar commented on howto/use-w3c-tracecontext/README.md:
```diff
@@ -0,0 +1,243 @@
+As mentioned earlier in the [section](#Using-Trace-Context-in-Dapr), you can create the trace context and pass it through when calling Dapr or Dapr can generate trace context passed it back to you. 
+
+If you choose to pass the trace context explictly, then Dapr will use the passed trace context and propagate all across the HTTP/gRPC call.
+We are using the same [grpc app](../create-grpc-app), however listing out the complete code for putting it all together.
```
Please rephrase to remove all first person references ("we", "I")

orizohar commented on howto/use-w3c-tracecontext/README.md:
```diff
@@ -0,0 +1,243 @@
+
+That's it !. Now you can correlate the calls in your app and in Dapr across services using the same trace context.
+
+To view traces, you can refer [how-to-diagnose-with-tracing](../diagnose-with-tracing) e.g Zipkin/Application Insights. 
```
This should be a "related links" section so it's more readable. Please add bullets for related links:
- general "observability" 
- observability sample
- Zipkin how-to
- repeat the W3C tracing context spec link here so it is easy to find again

orizohar commented on howto/use-w3c-tracecontext/README.md:
```diff
@@ -0,0 +1,243 @@
+
+This transformation of modern applications calls for a distributed tracing context propagation standard.
+
+W3C trace context specification defines a universally agreed-upon format for the exchange of trace context propagation data - referred to as trace context. Trace context solves the problems described above by
```
The link to the context (which is provided further down) might be better placed on the words "W3C trace context" here

orizohar commented on howto/use-w3c-tracecontext/README.md:
```diff
@@ -0,0 +1,243 @@
+# W3C Trace Context - Background
+
+Distributed tracing is a methodology implemented by tracing tools to follow, analyze and debug a transaction across multiple software components. Typically, a distributed trace traverses more than one component which requires it to be uniquely identifiable across all participating systems. Trace context propagation passes along this unique identification. Today, trace context propagation is imple ...
```
This section seems to explain what a trace context is. This is great but the sub section should be named accordingly.

shalabhms replied:
done.

orizohar commented on howto/use-w3c-tracecontext/README.md:
```diff
@@ -0,0 +1,243 @@
+# W3C Trace Context - Background
+
+Distributed tracing is a methodology implemented by tracing tools to follow, analyze and debug a transaction across multiple software components. Typically, a distributed trace traverses more than one component which requires it to be uniquely identifiable across all participating systems. Trace context propagation passes along this unique identification. Today, trace context propagation is imple ...
```
This article has several sections, please add a "contents" section with links to the sub headers for easy navigation

shalabhms replied:
done.

shalabhms commented:
> @shalabhms - this is very well written! On top of my other comments, please add to this PR changes to other docs that will link into this one (in their respective related links sections) such as:
> 
> * Main How To README - https://github.com/dapr/docs/tree/master/howto#observerability
> * Observability sample
> * Zipkin How To
> * Azure Monitor tracing How To
> * Observability  concept https://github.com/dapr/docs/blob/master/concepts/observability/README.md
> * Tracing concept https://github.com/dapr/docs/blob/master/concepts/observability/traces.md


Thanks @orizohar :-) . I have addressed the comments.

orizohar commented:
Thanks for addressing the comments. Just missing a link from main HowTo README https://github.com/dapr/docs/tree/master/howto#observerability
